src/core/reducers/categories.js

```diff
@@ -96,6 +104,47 @@ export function createEmptyCategoryList(): EmptyCategoryListType {
   }, {});
 }
 
+export const getCategories = (
+  categoriesState: CategoryMapType,
+  appName: string,
+  addonType: string,
+): Array<mixed> | null => {
+  if (
+    categoriesState &&
+    categoriesState[appName] &&
+    categoriesState[appName][addonType]
+  ) {
+    return Object.values(categoriesState[appName][addonType]);
+  }
+  return null;
+};
+
+export function getCategoryNames(
+  categoriesState: CategoryMapType,
+  addonCategories: AddonCategories,
+  appName: string,
+  addonType: string,
+): Array<mixed> | null {
+  invariant(categoriesState, 'categories state can not be empty!');
+  invariant(addonCategories, 'slugs can not be empty!');
+  invariant(appName, 'app name can not be empty!');
+  invariant(addonType, 'addon type can not be empty!');
+
+  const categories = getCategories(categoriesState, appName, addonType);
+
+  const relatedCategories = [];
+
+  if (categories && addonCategories && addonCategories[appName]) {
+    categories.forEach((r) => {
```

Comment:
We tend not to use short variables, like `r`, even in a loop. This might be better as `category`.

Reply:
I will replace `r` with `category`.